analysis/parameter_tuning.py

```diff
@@ -130,47 +129,82 @@ def _find_candidate_parameters(
     """
     calculate_l0_param = parameters_to_tune.max_partitions_contributed
     generate_linf = metric == pipeline_dp.Metrics.COUNT
+    generate_max_sum_per_partition = metric == pipeline_dp.Metrics.SUM
     calculate_linf_param = (parameters_to_tune.max_contributions_per_partition
                             and generate_linf)
+    calculate_sum_per_partition_param = (
+        parameters_to_tune.max_sum_per_partition and
+        generate_max_sum_per_partition)
     l0_bounds = linf_bounds = None
+    max_sum_per_partition_bounds = min_sum_per_partition_bounds = None
+
+    if calculate_sum_per_partition_param:
+        assert not parameters_to_tune.min_sum_per_partition, "Tuning of min_sum_per_partition is not supported yet"
+        assert hist.linf_sum_contributions_histogram.bins[
+            0].lower >= 0, "max_sum_per_partition should not contain negative sums because min_sum_per_partition tuning is not supported yet and therefore tuning for max_sum_per_partition works only when linf_sum_contributions_histogram does not negative sums"
```

Comment:
Nit: I think that `assert` here is too strong :). Let's make `logging.warning` ([link](https://docs.python.org/3/library/logging.html))